addons/account_update_tax_tags/tests/test_account_update_tax_tags_wizard.py

```diff
@@ -0,0 +1,333 @@
+        self.env.user.company_id = company_2
+        tax_2 = self._create_tax(
+            'update_test_tax_2',
+            tag_names=[None, 'update_test_invoice_tax_tag_company_2', None, None]
+        )
+        move_2 = self._create_move(tax_ids=tax_2)
+        self._change_tax_tag(tax_2, 'invoice_tax_tag_changed_for_company_2', invoice=True, base=False)
+        self.wizard.update_amls_tax_tags()
+
+        tax_line_1 = move_1.line_ids.filtered(lambda aml: aml.tax_line_id)
+        tax_line_2 = move_2.line_ids.filtered(lambda aml: aml.tax_line_id)
+        # Only the first move_id should be updated since it belongs to first company which was initialized in the wizard
+        self.assertEqual(tax_line_1.tax_tag_ids.name, 'invoice_tax_tag_changed_for_company_1')
+        self.assertEqual(tax_line_2.tax_tag_ids.name, 'update_test_invoice_tax_tag_company_2')
+
+    def test_update_all_move_type(self):
+        """ Tests that all move type are correctly updated with the corresponding tags. """
+        move_types = [
+            # 'entry' tested in test_update_move_type_entry since it's more complex
+            'out_invoice',
+            'in_invoice',
+            'out_refund',
+            'in_refund',
+            'in_receipt',
+            'out_receipt',
+        ]
+        for move_type in move_types:
+            for line_type in ['base', 'tax']:
+                with self.subTest(f'Update tax tag on {move_type}-{line_type}'):
+                    tax_2 = self._create_tax('update_test_tax_2', tag_names=[
+                        'test_tag_invoice_base',
+                        'test_tag_invoice_tax',
+                        'test_tag_refund_base',
+                        'test_tag_refund_tax'
+                    ])
+                    move = self._create_move(move_type=move_type, tax_ids=tax_2)
+                    super_type = move_type.split('_')[1]
+                    if super_type == 'receipt':
+                        super_type = 'invoice'  # receipt type acts just like invoice one
+                    self._change_tax_tag(tax_2, f'{move_type}_{line_type}_tag_changed', invoice=super_type == 'invoice', base=line_type == 'base')
+                    self.wizard.update_amls_tax_tags()
+
+                    invoice_line, tax_line, counterpart_line = self._get_amls_by_type(move)
+                    if line_type == 'base':
+                        self.assertEqual(invoice_line.tax_tag_ids.name, f'{move_type}_{line_type}_tag_changed', 'Invoice lines tags should have changed.')
+                        self.assertEqual(tax_line.tax_tag_ids.name, f'test_tag_{super_type}_tax', 'Tax lines tags should not have changed.')
+                        self.assertFalse(counterpart_line.tax_tag_ids, 'Counterpart lines tags should not have changed.')
+                    else:
+                        self.assertEqual(invoice_line.tax_tag_ids.name, f'test_tag_{super_type}_base', 'Invoice lines tags should not have changed.')
```

Comment:
subTest will not rollback (I think), so here, even if your base tags were changed by the wizard while they shouldn't, they will have the same value as the original ones. So essentially, this check will always be true. It would be nice to fix this hole in the test coverage. Maybe just generating tag_names containing the line_type, so that they change at each iteration ? You could even put the move_type in them as well